src/server/db/mongo/user-privat-provider.js

```diff
@@ -0,0 +1,157 @@
+import UserPrivateModel from './models/user-private.js'
+import logger from '../../../imports/pino-logger'
+import { type UserRecord } from '../../../imports/types'
+
+class UserPrivate {
+  constructor() {
+    this.model = UserPrivateModel
+  }
+
+  /**
+   * Get new nonce after increment
+   *
+   * @param {UserRecord} user
+   *
+   * @returns {Promise<*>}
+   */
+  async isDupUserData(user: UserRecord): boolean {
+    const { email, mobile } = user
+    let result = null
+
+    if (email) {
+      result = await this.model.findOne({ email })
+      if (result) {
+        return true
+      }
+    }
+
+    if (mobile) {
+      result = await this.model.findOne({ mobile })
+      if (result) {
+        return true
+      }
+    }
+
+    return false
+  }
+
+  /**
+   * Create or update user privat date
+   *
+   * @param {UserRecord} user
+   *
+   * @returns {Promise<void>}
+   */
+  async updateUser(user: UserRecord): Promise<boolean> {
+    try {
+      const userDb = await this.getByIdentifier(user.identifier)
+      if (userDb) {
+        await this.model.update({ identifier: user.identifier }, { $set: user }, { new: true })
```

Comment:
use updateOne with upsert:true instead of new:true
If we upsert why do we need to fetch and check if userDb exists?

Reply:
done